playbooks/prepare_stack_testconfig.yaml

```diff
@@ -0,0 +1,92 @@
+---
+- hosts: standalone
+  become: true
+  become_user: stack
+  gather_facts: false
+  vars_files: vars/defaults.yaml
+
+  tasks:
+
+  - name: Gather host environment variables
+    setup:
+      filter: ansible_env
+
+  - name: Ensure remote .ssh directory
+    file:
+      path: "{{ ansible_env.HOME }}/.ssh"
+      state: directory
+      owner: stack
+      group: stack
+      mode: '0700'
+
+  - name: Copy ssh public key to host
+    copy:
+      src: "{{ testconfig_public_key }}"
+      dest: "{{ ansible_env.HOME }}/.ssh/{{ testconfig_public_key | basename }}"
+      owner: stack
+      group: stack
+      mode: '0600'
+    register: testconfig_public_key_copy
+
+  - name: Configure resources for openshift user
```

Comment:
ideally we would use ```OpenStack Cloud``` collections, e.g. [keypair](https://docs.ansible.com/ansible/latest/collections/openstack/cloud/keypair_module.html#ansible-collections-openstack-cloud-keypair-module), etc. but it's just an FYI.

Reply:
Agreed. I've had a poor experience with those in the past, though, a combination of being both buggy and incomplete. On a previous project I started using them exclusively, then switched to cli for the stuff which didn't work or wasn't possible, then decided the complication of supporting 2 methods wasn't worth it and switched to cli for everything.

Ideally I'd go all in and just fix/improve the ansible. I never seem to be working on something where that seems worthwhile, although I'm aware that's short term thinking.